internal/utils.go

```diff
@@ -120,42 +121,45 @@ func getManifests(policyConf *types.PolicyConfig) ([]map[string]interface{}, err
 // that each template includes a single manifest specified in policyConf.
 // An error is returned if one or more manifests cannot be read or are invalid.
 func getPolicyTemplates(policyConf *types.PolicyConfig) ([]map[string]map[string]interface{}, error) {
-	manifests, err := getManifests(policyConf)
+	manifestGroups, err := getManifests(policyConf)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(manifests) == 0 {
-		return nil, fmt.Errorf(
-			"the policy %s must specify at least one non-empty manifest file", policyConf.Name,
-		)
-	}
-
-	objectTemplatesLength := len(manifests)
+	objectTemplatesLength := len(manifestGroups)
 	policyTemplatesLength := 1
 	if !policyConf.ConsolidateManifests {
-		policyTemplatesLength = len(manifests)
+		policyTemplatesLength = len(manifestGroups)
 		objectTemplatesLength = 0
 	}
 	objectTemplates := make([]map[string]interface{}, 0, objectTemplatesLength)
 	policyTemplates := make([]map[string]map[string]interface{}, 0, policyTemplatesLength)
-	for _, manifest := range manifests {
-		objTemplate := map[string]interface{}{
-			"complianceType":   policyConf.ComplianceType,
-			"objectDefinition": manifest,
-		}
-		if policyConf.ConsolidateManifests {
-			// put all objTemplate with manifest into single consolidated objectTemplates object
-			objectTemplates = append(objectTemplates, objTemplate)
-		} else {
-			// casting each objTemplate with manifest to objectTemplates type
-			// build policyTemplate for each objectTemplates
-			policyTemplate := buildPolicyTemplate(policyConf, &[]map[string]interface{}{objTemplate})
-			setNamespaceSelector(policyConf, policyTemplate)
-			policyTemplates = append(policyTemplates, *policyTemplate)
+	for i, manifestGroup := range manifestGroups {
+		complianceType := policyConf.Manifests[i].ComplianceType
+		for _, manifest := range manifestGroup {
```

Comment:
I believe we could flat this nested loop but lets keep it as enhancement for future work